src/Pages/TabPage.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useState, useEffect } from 'react';
+import cn from 'classnames';
+import { Link, useParams } from 'react-router-dom';
+import { Tab } from '../types/Tab';
+import { Navbar } from '../components/Navbar';
+
+const tabs: Tab[] = [
+  { id: 'tab-1', title: 'Tab 1', content: 'Some text 1' },
+  { id: 'tab-2', title: 'Tab 2', content: 'Some text 2' },
+  { id: 'tab-3', title: 'Tab 3', content: 'Some text 3' },
+];
+
+export const TabPage: React.FC = () => {
+  const [selectedTab, setSelectedTab] = useState<Tab | undefined>(undefined);
+  const { tabId } = useParams();
+
+  useEffect(() => {
+    const selected = tabs.find(tab => tab.id === tabId);
+
+    setSelectedTab(selected);
+  }, [tabId]);
```

Comment:
I think you dont need useEffect here. So you dont need state. I would prefer:
```
const selectedTabId = useMemo(
  tabs.find(tab => tab.id === tabId)?.content, [tabId]
)
```

Reply:
agree with @VladimirEdelmann